### ISSUE TITLE
refactor: rename zero_tolerance to patience for clarity

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/rbig#60]
## Problem

The `zero_tolerance` parameter in `AnnealedRBIG` is confusingly named. It sounds like a tolerance value (a threshold), but it's actually a **patience counter** — the number of consecutive layers with TC change below `tol` before early-stopping.

The actual tolerance threshold is controlled by the separate `tol` parameter.

## Current behavior

```python
model = AnnealedRBIG(
    n_layers=100,
    zero_tolerance=60,  # ← sounds like a threshold, but it's a layer count
    tol=1e-5,           # ← this is the actual threshold
)
```

The default `zero_tolerance=60` means the model fits **60 extra layers** after convergence before stopping. This is extremely conservative and makes fitting much slower than necessary.

## Proposed changes

1. **Rename** `zero_tolerance` → `patience` (consistent with ML conventions like Keras, PyTorch Lightning, etc.)
2. **Reduce default** from 60 → 10 (60 extra layers of wasted compute is excessive)
3. **Deprecation path**: Accept both names for one release cycle, warn on old name
4. **Update docstrings** to clearly explain that `patience` is a count of consecutive non-improving layers, not a threshold

## Example after change

```python
model = AnnealedRBIG(
    n_layers=100,
    patience=10,   # stop after 10 consecutive layers with |ΔTC| < tol
    tol=1e-5,      # threshold for "meaningful" TC improvement
)
```

## Labels

enhancement, api
